tests/unit/remoteworker/test_remote_worker.py

```diff
@@ -149,6 +152,161 @@ def test_return_url_per_environment(self):
+                b"Dummy file content",
+                content_type="text/plain"
+            ),
+            "method_name": "Test Method",
+            "method_description": "Test Description",
+            "project_url": "http://testserver",
+            "publication_url": "http://testserver",
+            "is_public": True
+        }
+
+    def tearDown(self):
+        try:
+            shutil.rmtree("/tmp/evalai")
+        except OSError:
+            pass
+
+    @mock.patch("scripts.workers.remote_submission_worker.SUBMISSION_DATA_DIR", "mocked/dir/submission_{submission_id}")
+    @mock.patch("scripts.workers.remote_submission_worker.os.path.basename", return_value="user_annotation_file.txt")
+    @mock.patch("scripts.workers.remote_submission_worker.get_submission_by_pk")
+    @mock.patch("scripts.workers.remote_submission_worker.get_challenge_by_queue_name")
+    @mock.patch("scripts.workers.remote_submission_worker.get_challenge_phase_by_pk")
+    @mock.patch("scripts.workers.remote_submission_worker.create_dir_as_python_package")
+    @mock.patch("scripts.workers.remote_submission_worker.download_and_extract_file")
+    @mock.patch("scripts.workers.remote_submission_worker.run_submission")
+    def test_process_submission_message_success(self, mock_run_submission,
+                                                mock_download_and_extract_file,
+                                                mock_create_dir_as_python_package,
+                                                mock_get_challenge_phase_by_pk,
+                                                mock_get_challenge_by_queue_name,
+                                                mock_get_submission_by_pk,
+                                                mock_basename):
+        message = {
+            "challenge_pk": self.challenge_pk,
+            "phase_pk": self.challenge_phase_pk,
+            "submission_pk": self.submission_pk
+        }
+        user_annotation_file_path = "mocked/dir/submission_{}/user_annotation_file.txt".format(self.submission_pk)
+
+        with mock.patch("scripts.workers.remote_submission_worker.extract_submission_data") as mock_esd:
+            process_submission_message(message)
+            mock_esd.assert_called_with(self.submission_pk)
+
+        mock_get_submission_by_pk.return_value = self.submission
+        mock_get_challenge_by_queue_name.return_value = self.challenge
+        mock_get_challenge_phase_by_pk.return_value = self.challenge_phase
+
+        process_submission_message(message)
+
+        mock_run_submission.assert_called_with(
+            self.challenge_pk,
+            self.challenge_phase,
+            self.submission,
+            user_annotation_file_path,
+            None
+        )
+
+    @mock.patch("scripts.workers.remote_submission_worker.extract_submission_data")
+    def test_process_submission_message_when_submission_does_not_exist(self, mock_esd):
+        self.submission_pk += 1
```

Comment:
@takitsuse rather than incrementing the actual variable can you use `self.submission_pk + 1` wherever it is needed. As this is a test for invalid submission we shouldn't update the actual  `self.submission_pk`